R/simulate_susceptibles.R

```diff
@@ -0,0 +1,165 @@
+##' Simulate a single chain using a branching process while accounting
+##' for depletion of susceptibles.
+##'
+##' @param offspring offspring distribution: a character string corresponding to
+##'   the R distribution function. Currently only "pois" & "nbinom" are
+##'   supported. Internally truncated distributions are used to avoid infecting
+##'   more people than susceptibles available.
+##' @param mn_offspring the average number of secondary cases for each case
+##' @param disp_offspring the dispersion coefficient (var/mean) of the number of
+##'      secondary cases. Ignored if offspring == "pois". Must be > 1.
+##' @param serial the serial interval. A function that takes one parameter
+##'     (`n`), the number of serial intervals to randomly sample.
+##'     Value must be >= 0.
+##' @param t0 start time (if serial interval is given); either a single value (0
```

Comment:
For parameters that are exactly the same (with the same description) as in `chain_sim` you could remove the documentation here and instead insert a line `##' @inheritParams chain_sim` below